modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -974,11 +975,19 @@ impl Wallet {
 
             // TODO: use batching for mainnet syncing
             trace!(block = height, "Fetching block hash");
-            let block_hash = self
-                .btc_rpc
-                .get_block_hash(height as u64)
+            let block_hash =
+                retry(
+                    "get_block_hash",
+                    FibonacciBackoff::default()
+                        .with_min_delay(Duration::from_secs(1))
+                        .with_max_delay(Duration::from_secs(10 * 60))
+                        .with_max_times(usize::MAX),
+                    || {
+                        self.btc_rpc.get_block_hash(height as u64) // TODO: use u64 for height everywhere
```

Comment:
Why do we need u64? 4B blocks is going to get a while to get to?

Reply:
Why so bearish on the longevity of the chain :smiley:

I'm preserving the original comment and created an issue a while ago that surfaces some of my reasoning why u64 makes sense. Feel free to push back on the issue.

https://github.com/fedimint/fedimint/issues/3513